doobie/src/main/scala/tamer/db/TamerDoobieJob.scala

```diff
@@ -0,0 +1,89 @@
+package tamer.db
+
+import com.sksamuel.avro4s.Codec
+import doobie.Query0
+import doobie.implicits.{toDoobieStreamOps, _}
+import doobie.util.transactor.Transactor
+import fs2.Stream
+import log.effect.LogWriter
+import log.effect.zio.ZioLogWriter.log4sFromName
+import tamer.{TamerError, db}
+import tamer.config.{Config, KafkaConfig}
+import tamer.db.Compat.toIterable
+import tamer.db.ConfigDb.{DbConfig, QueryConfig}
+import tamer.job.{AbstractTamerJob, SourceStateChanged}
+import zio.blocking.Blocking
+import zio.interop.catz.taskConcurrentInstance
+import zio.{Chunk, Layer, Queue, Ref, Schedule, Task, UIO, ZEnv, ZIO}
+
+import java.time.{Duration, Instant}
+
+object TamerDoobieJob {
+  def apply[
+      R <: ZEnv with DbConfig with TamerDBConfig with KafkaConfig,
+      K <: Product: Codec,
+      V <: Product: Codec,
+      S <: Product: Codec
+  ](
+      setup: DoobieConfiguration[K, V, S]
+  ) = new TamerDoobieJob[R, K, V, S](setup)
+
+  final def fetchWithTimeSegment[K <: Product: Codec, V <: Product with Timestamped: Ordering: Codec](
+      queryBuilder: TimeSegment => Query0[V]
+  )(earliest: Instant, tumblingStep: Duration, keyExtract: V => K): ZIO[ZEnv, TamerError, Unit] = {
+    val transactorLayer: Layer[TamerError, DbTransactor]               = (Blocking.live ++ ConfigDb.live) >>> db.hikariLayer
+    val queryConfigLayer: Layer[TamerError, DbConfig with QueryConfig] = ConfigDb.live
+    val setup: DoobieConfiguration[K, V, TimeSegment]                  = DoobieConfiguration.fromTimeSegment[K, V](queryBuilder)(earliest, tumblingStep, keyExtract)
+    apply(setup).fetch().provideSomeLayer[ZEnv](transactorLayer ++ queryConfigLayer ++ Config.live)
+  }
+
+}
+class TamerDoobieJob[
+    R <: ZEnv with DbConfig with TamerDBConfig with KafkaConfig,
+    K <: Product: Codec,
+    V <: Product: Codec,
+    S <: Product: Codec
+](
+    setup: DoobieConfiguration[K, V, S]
+) extends AbstractTamerJob[R, K, V, S, Unit](setup.generic) {
+  import eu.timepit.refined.auto._
+
+  private[this] final val logTask: Task[LogWriter[Task]] = log4sFromName.provide("tamer.db")
+
+  override protected def createInitialSourceState: Unit = ()
+
+  override protected def createSchedule: Schedule[Any, Any, (Duration, Long)] =
+    Schedule.recurs(0).map(_ => (Duration.ZERO, 0))
+
+  override protected def updatedSourceState(currentState: Ref[Unit], token: Queue[Unit]): ZIO[R, Throwable, SourceStateChanged] =
+    ZIO.succeed(SourceStateChanged(false))
```

Comment:
wouldn't this suggest that these attributes don't belong to `AbstractTamerJob` but are really used in the S3 module only? maybe there's space for another abstraction that contains them, say in case we want/need an objectstorage version that behaves like s3